File: back/src/main/java/com/back/domain/review/validation/ValidTags.java
package com.back.domain.review.validation;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.FIELD;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Target(FIELD)
@Retention(RUNTIME)
@Constraint(validatedBy = ValidTagsValidator.class)
public @interface ValidTags {
    String message() default "tags는 최대 5개, 각 tag는 서로 중복되지 않고 1~20자여야 합니다.";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}

File: back/src/main/kotlin/com/back/domain/review/validation/ValidTags.kt
package com.back.domain.review.validation

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [ValidTagsValidator::class])
annotation class ValidTags(
    val message: String = "tags는 최대 5개, 각 tag는 서로 중복되지 않고 1~20자여야 합니다.",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)